tests/onnx/test_onnx_model_export.py

```diff
@@ -317,28 +289,19 @@ def test_pyfunc_representation_of_float32_model_casts_and_evalutes_float64_input
         atol=1e-05,
     )
 
-    with pytest.raises(RuntimeError):
+    with pytest.raises(Exception, match="Unexpected input data type"):
```

Comment:
The ONNX runtime no longer raises a `RuntimeError`. I considered removing this case along with casting behavior as discussed in the TODO comment on line 266, but it's not clear to me that https://github.com/mlflow/mlflow/issues/1286 was truly fixed (it appears to have been closed). I'm also not sure that tensor input schemas will fully address this use case. Accordingly, I've opted to keep the casting behavior and associated test case for now.